services/skus/model/model.go

```diff
@@ -126,6 +143,98 @@ func (o *Order) CreateStripeCheckoutSession(
 	return CreateCheckoutSessionResponse{SessionID: session.ID}, nil
 }
 
+// CreateRadomCheckoutSession creates a Radom checkout session for o.
+func (o *Order) CreateRadomCheckoutSession(
+	ctx context.Context,
+	client radomClient,
+	sellerAddr string,
+) (CreateCheckoutSessionResponse, error) {
+	return o.CreateRadomCheckoutSessionWithTime(ctx, client, sellerAddr, time.Now().Add(24*time.Hour))
+}
+
+// CreateRadomCheckoutSessionWithTime creates a Radom checkout session for o.
+func (o *Order) CreateRadomCheckoutSessionWithTime(
+	ctx context.Context,
+	client radomClient,
+	sellerAddr string,
+	expiresAt time.Time,
+) (CreateCheckoutSessionResponse, error) {
+
+	// get the environment so we know what is acceptable chain/tokens
+	gateway := radom.Gateway{
+		Managed: radom.Managed{
+			Methods: []radom.Method{},
+		},
+	}
+	env, ok := ctx.Value(appctx.EnvironmentCTXKey).(string)
+	if !ok || env != "production" {
+		// append testnet
+		gateway.Managed.Methods = append(gateway.Managed.Methods, radom.Method{
+			Network: "SepoliaTestnet",
+			Token:   "0x5D684d37922dAf7Aa2013E65A22880a11C475e25",
+		}, radom.Method{
+			Network: "PolygonTestnet",
+			Token:   "0xd445cAAbb9eA6685D3A512439256866563a16E93",
+		})
+	} else {
+		// append production
+		gateway.Managed.Methods = append(gateway.Managed.Methods, radom.Method{
+			Network: "Polygon",
+			Token:   "0x3cef98bb43d732e2f285ee605a8158cde967d219",
+		}, radom.Method{
+			Network: "Ethereum",
+			Token:   "0x0d8775f648430679a709e98d2b0cb6250d2887ef",
+		})
+	}
```

Comment:
Non-blocking: Ideally, these should take place after we've done the validation steps below, but before making a request. I.e. there is little point in initialising this if the order itself is not worth action on.

Reply:
I moved gateway management inside client. Upon a client method call (create session), the client will determine the appropriate gateway settings and attach them to the request.

This knowledge about the gateway does not belong in the order context. It is also not different from order to order. That means it does not have to be here, and it has to be an internal to the client detail.

Plus, allocating slices for every single order call is not necessary, and we can reuse the same gateway methods again and again.

Pushed that change.